src/renderer/src/views/MigratingProject/index.tsx

```diff
@@ -0,0 +1,175 @@
+import { Button, Container, LinearProgress, Typography, useTheme } from '@mui/material'
+import CheckIcon from '@mui/icons-material/CheckCircleRounded'
+import { Column, Row } from '@renderer/components/LayoutComponents'
+import { Logo, Mapeo5Logo } from '@renderer/components/Logo'
+import { DefaultLayout } from '@renderer/layouts/default'
+import { useEffect, useState } from 'react'
+import { defineMessages, useIntl } from 'react-intl'
+import { useLocation } from 'react-router-dom'
+
+const INTERVAL_DURATION = 250
+
+type LocationState = { observations: number; media: number }
+
+export const MigratingProjectView = () => {
+  const theme = useTheme()
+  const { state } = useLocation()
+  const locationState = state as LocationState
+  const intl = useIntl()
+  const observationsTotal = locationState.observations
+  const mediaTotal = locationState.media
+  const total = observationsTotal + mediaTotal
+  const [observationsMigrated, setObservationsMigrated] = useState(0)
+  const [mediaMigrated, setMediaMigrated] = useState(0)
+  const [progress, setProgress] = useState(0)
+  let interval: NodeJS.Timer
+
+  const getProgress = () => ((mediaMigrated + observationsMigrated) / total) * 100
+
+  useEffect(() => {
+    interval = setInterval(updateProgress, INTERVAL_DURATION)
+
+    return () => {
+      clearInterval(interval)
+    }
+  }, [])
+
+  useEffect(() => {
+    setProgress(getProgress())
+
+    console.log({ progress })
+
+    if (observationsMigrated === observationsTotal && mediaMigrated === mediaTotal) {
+      clearInterval(interval)
+    }
+  }, [observationsMigrated, observationsTotal, mediaMigrated, mediaTotal])
+
+  const updateProgress = () => {
+    const obsMigrating = Math.floor(Math.random() * 6)
+    const mediaMigrating = Math.floor(Math.random() * 6)
+
+    setObservationsMigrated((prevObservationsMigrated) =>
+      prevObservationsMigrated + obsMigrating < observationsTotal
+        ? prevObservationsMigrated + obsMigrating
+        : observationsTotal,
+    )
+    setMediaMigrated((prevMediaMigrated) =>
+      prevMediaMigrated + mediaMigrating < mediaTotal ? prevMediaMigrated + mediaMigrating : mediaTotal,
+    )
+  }
+
```

Comment:
This can be simplified by just deriving `progress`:

```ts

  const updateProgress = useCallback(() => {
    const obsMigrating = Math.floor(Math.random() * 6)
    const mediaMigrating = Math.floor(Math.random() * 6)

    setObservationsMigrated((prevObservationsMigrated) =>
      prevObservationsMigrated + obsMigrating < observationsTotal
        ? prevObservationsMigrated + obsMigrating
        : observationsTotal,
    )
    setMediaMigrated((prevMediaMigrated) =>
      prevMediaMigrated + mediaMigrating < mediaTotal ? prevMediaMigrated + mediaMigrating : mediaTotal,
    )
  },[])

  useEffect(() => {
    const interval = setInterval(updateProgress, INTERVAL_DURATION)

    return () => {
      clearInterval(interval)
    }
  }, [updateProgress])

  const progress = ((mediaMigrated + observationsMigrated) / total) * 100
```

Reply:
For Reference: https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state

Reply:
ah yeah, seems obvious in retrospect. Thanks!